### ISSUE TITLE
Can one configure Timetraveler for multistep prediction?

### ISSUE THREAD

**Issue by JuliaGast** [JHL-HUST/TITer#3]
Hello

**Terminology:**
With _single step_ prediction I mean, when testing on the test set to always predict one step in the future, then feed the ground truth, and predict another step. This is what I understand you do with Timetraveler.
With _multistep_ prediction I mean to predict all steps from the test set in the future, without feeding the ground truth in between. Thus, the model will never get fed any triples or quadruples from the test set for their prediction. This is what some other works, like RE-NET (https://arxiv.org/abs/1904.05530) do.

**Why I understand Timetraveler does single-step prediction:**
When running main.py –do_test:
When calling tester.test (main.py: l. 215):
When returning the actions (with `get_padd_actions()`) you call  environment.py `get_state_actions_space_complete()`.
This will return an action array, containing (as far as I understand)  for a certain query at t, the neighbors at all timesteps t_minus <t. 
However, it can be, that these timesteps t_minus are timesteps in the test set.
Here is a screenshot of an example (every line with comment #added Julia has been added by my, for illustration) 
	In environment.py get_state_actions_space_complete(), you do the following:
 
![image](https://user-images.githubusercontent.com/41183430/183965678-6c723412-5636-4491-9755-6aaa1f0003a3.png)
![image](https://user-images.githubusercontent.com/41183430/183965700-b338eaa7-8ee5-4784-b6f3-89db057189c0.png)
 
This means, **you return quadruples from the test set** (that are smaller than the current timestep time) as potential actions. This is what I consider single step prediction.

**Questions:**

1. Have you ever tried Timetraveler for multistep prediction? If yes, is it possible to set some parameter, or could you please provide some code modifications, so that I can also try this?
•	I assume, one thing that one could do would be to modify in preprocess_data, the data that is used as input to the state_actions_space:
•	Instead of using dataset.allQuadruples one could use dataset.trainQuadruples  (and dataset.validQuadruples) (line 32 – 39)
•	This would mean, that in environment.py get_state_actions_space_complete() only test quadruples from the train and validation set can be used, and none from the test set 
•	Does this make sense, is or am I mistaken? Have you tried this?
•	Are there any other modifications in the code necessary that I am not aware of? 

2. Is it even possible to do multistep prediction, or is Timetraveler not made for this setting at all?

3. Did you compare, in your paper, to RE-NET multistep prediction (ie the default setting of Renet)? If no: can you please describe how you did compare to singlestep prediction? The same question is true for e.g. Cy-GNet.

Thank you very much for your response.
Julia
